### PR TITLE
Fix support model versions

### LINKED ISSUE #141: precommit fails on different model versions
**Describe the bug**
- dbt `>1.5` introduced [model versions](https://docs.getdbt.com/docs/collaborate/govern/model-versions)
- the precommit hooks `Check the model has properties file` and `Check the model has description` fail
- probably because it expects description for each version, while there is none

**To Reproduce**
1. Create two versions of the same model `fancy_model_v1.sql` and `fancy_model_v2.sql` 
2. Adapt the `yml` file
```yml
- name: fancy_model
  latest_version: 1
  description: fancy model
  config:
    contract:
      enforced: true
  columns:
    - name: tool_raw
      description: raw name of tool
      data_type: varchar(20)
      constraints:
        - type: not_null
        - type: primary_key
          warn_unenforced: False
    - name: tool
      description: tool name, "jpg-to-pdf"
      data_type: varchar(20)

  versions:
    - v: 1
    - v: 2
      columns:
        - include: all
          exclude: [tool]
        - name: new_column
          data_type: varchar(6)
          description: "I am an experimental column"
```
3. Run the `Check the model has properties file` and `Check the model has description`
4. The error is:
```
#14 [dbt_precommit 4/4] RUN pre-commit run --all-files
...
#14 1.186 Check the model has properties file......................................Failed
#14 8.483 - hook id: check-model-has-properties-file
#14 8.483 - exit code: 1
#14 8.483 models/fancy_model_v1.sql: does not have model properties defined in any .yml file.
#14 8.483 models/fancy_model_v2.sql: does not have model properties defined in any .yml file.
...
#14 8.484 Check the model has description..........................................Failed
#14 34.62 - hook id: check-model-has-description
...
#14 34.62 models/fancy_model_v1.sql: does not have defined description or properties file is missing.
#14 34.62 models/fancy_model_v2.sql: does not have defined description or properties file is missing.
```

**Expected behavior**
- `precommit` should understand that  `fancy_model_v1.sql` and `fancy_model_v2.sql` are the same and have description under `fancy_model` in `yml` file

**Version:**
v0.1.0

**Additional context**

### PR DESCRIPTION
**Pull request by gbrunois**
This PR fix #141 

- Update `utils.py` to support model version
- Test a node with a version

### CODE REVIEW

**Comment by BAntonellini** [comment]
Good one @gbrunois, thank you for contributing

**Comment by awal11** [comment]
I am struggling with some tests (eg check-model-has-tests) not finding anything - and it seems to be related to the fact that we are using versioned models, so this patch could help.

I just tried to run it directly from pre-commit run --all-files , pointing the repo to main branch

but it fails with:

>  File "C:\Users\user1\.cache\pre-commit\repohuje0l2b\py_env-python3.10\lib\site-packages\dbt_checkpoint\check_model_parents_schema.py", line 85, in main
>     status_code = check_parents_schema(
> 
>   File "C:\Users\user1\.cache\pre-commit\repohuje0l2b\py_env-python3.10\lib\site-packages\dbt_checkpoint\check_model_parents_schema.py", line 33, in check_parents_schema
>     for model in models:
> 
>   File "C:\Users\user1\.cache\pre-commit\repohuje0l2b\py_env-python3.10\lib\site-packages\dbt_checkpoint\utils.py", line 167, in get_models
> 
> if node.get("version") and split_key[-1] == "v" + node.get("version"):
> 
> TypeError: can only concatenate str (not "int") to str
>

**Comment by BAntonellini** [comment]
Good catch @awal11. Model versions are indeed stored as integers in dbt Manifest.

Could you re-run and let us know?

**Comment by awal11** [comment]
this worked partially - I mean there is no more TypeError, check-model-has-tests passes.... but it should not, it still somehow ignores all these versioned models that have insufficient number of tests. 

I will try to dig deeper into it. Is there a way to force dbt-checkpoint to provide some debugging information, showing which parameters it received and which models is it processing? I was looking but cannot find anything.

**Comment by awal11** [comment]
ok, i think I understand that happens:

1. utils.get_model_sqls returns a map of filenames, and each returned filename contains the **_vx** at the end: person**_v1**
2. utils.get_model currently removes the ending from the filename and the condition below is not met
` if filename in filenames and split_key[0] == "model": `

Locally I patched the filename assignment and it seems to work: 
```
        # Versions are supported since dbt-core 1.5
        if node.get("version") and split_key[-1] == f"v{node.get('version')}":
            filename = f"{split_key[-2]}_v{node.get('version')}"
        else:

```

File: tests/conftest.py [hunk 1]
```diff
@@ -15,6 +15,16 @@
         "adapter_type": "snowflake",
     },
     "nodes": {
+        "model.with_version.v1": {
+            "patch_path": "project://bb/with_version.yml",
+            "path": "aa/bb/with_version.sql",
```
Comment (awal11) [inline tests/conftest.py:20]: this path should be: "aa/bb/with_version_v1.sql"

File: tests/unit/test_check_model_has_properties_file.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 TESTS = (
     (["aa/bb/with_schema.sql", "--is_test"], True, True, 0),
     (["aa/bb/with_schema.sql", "--is_test"], False, True, 1),
+    (["aa/bb/with_version.sql", "--is_test"], True, True, 0),
```
Comment (awal11) [inline tests/unit/test_check_model_has_properties_file.py:8]: this path should be: "aa/bb/with_version_v1.sql"

**Comment by BAntonellini** [comment]
Great contribution @awal11, tested all out locally and it works fine now.

**Pull request merged**